internal/installer/elasticagent_tar_macos.go

```diff
@@ -215,7 +215,7 @@ func (i *elasticAgentTARDarwinPackage) Stop(ctx context.Context) error {
 
 // Uninstall uninstalls a TAR package
 func (i *elasticAgentTARDarwinPackage) Uninstall(ctx context.Context) error {
-	cmds := []string{"sudo", "elastic-agent", "uninstall", "-f"}
+	cmds := []string{"sudo", "./elastic-agent/elastic-agent", "uninstall", "-f"}
```

Comment:
This is not correct: the install command is executed from the extracted-from-tar binary, but the uninstall needs to be executed from the system-installed (in PATH) binary for the agent